src/globus_cli/commands/gcp/__init__.py

```diff
@@ -6,6 +6,10 @@
     lazy_subcommands={
         "create": (".create", "create_command"),
         "update": (".update", "update_command"),
+        "set-subscription-id": (
+            "endpoint.set_subscription_id",
+            "set_endpoint_subscription_id",
+        ),
```

Comment:
I'm okay with this as a short-term position, but I'd like us to write a new command for this, rather than reusing the existing one.

My reason is primarily helptext and parameter names. If you look at the existing command, you'll see it states
```
Unlike the '--managed' flag for 'globus endpoint update' ...
```

Which is not something we want to retain into the future.

I think it's also worth having the new command check to confirm that it's operating on a `GCP_mapped_collection` rather than a `GCP_guest_collection`. That way, we more clearly explain to users that it's only possible to use the relevant command on a GCP host, not the shares underneath that host.

I'm perfectly happy for us to pursue that change as a fast-follow to this PR though.

Reply:
I initially did write a new command for GCP but erased it when it ended up being identical to this one.

---

If we do add a duplicate code path, it's worth noting that introduces the risk of drift (one path gets updated but not the other).

If it's just a helptext concern, I don't think that's worth a copy/paste. If it's adding additional validation of `mapped` vs `guest` collection types; I could buy it. It's a fairly small copy/paste.

---

I'll include that in a new rev of this PR.

Reply:
I'm not that concerned about the paths being duplicated, since I think they are actually _different_. The question IMO is only how much it's worth working hard to avoid code duplication.

I want the following:
- GCP-specific checks in the GCP command
- legacy-specific deprecations in the legacy command
- different helptext for the two

The worry about drift would be more pronounced if this didn't take the form of a deprecation. However, I don't think we want to keep `globus endpoint` (the whole tree), so we should be making moves to deprecate its various pieces.